File: backend/app/schemas/screener.py
from pydantic import BaseModel


class ScreenerItem(BaseModel):
    ticker: str
    name: str
    market: str
    close: int | None = None
    change_pct: float | None = None
    volume: int | None = None
    volume_ratio: float | None = None
    momentum_score: float | None = None
    market_cap: int | None = None


class ScreenerResponse(BaseModel):
    items: list[ScreenerItem]
    total: int

File: frontend/src/types/screener.ts
export interface ScreenerItem {
  ticker: string
  name: string
  market: string
  close: number
  change_pct: number
  volume: number
  volume_ratio: number | null
  momentum_score: number | null
  market_cap: number | null
}

export interface ScreenerResponse {
  items: ScreenerItem[]
  total: number
}

export type ScreenerTab = 'gainers' | 'losers' | 'volume' | 'highs' | 'momentum'
